src/term.rs

```diff
@@ -586,7 +738,8 @@ impl Term {
             }
             Term::Array(elements, _) => {
                 let elements_str: Vec<String> = elements
-                    .iter()
+                    .clone()
```

Comment:
While I understand that you need to keep a Rc to the backing array while iterating, it's a bit disturbing as a code change to add a `clone()` here. Nitpick: I would encapsulate this function as an `iter()` function directly on `Array` (and I think that would gives us for loops iteration for free?)

Reply:
In fact, after re-reading, I think we don't have a proper `iter()` replacement. Here the iterator will clone each rich term upon iteration, if I'm not mistaken. Maybe we want a proper `&RichTerm` iterator for `Array`?